addons/website_slides/models/slide_channel.py

```diff
@@ -45,8 +45,17 @@ def _compute_completion(self):
             slide_done = mapped_data.get(record.channel_id.id, dict()).get(record.partner_id.id, 0)
             slide_total = channel_data.get(record.channel_id.id) or 1
             record.completion = math.ceil(100.0 * slide_done / slide_total)
-            if record.completion >= 100:
-                record.update({'completed': True})
+
+    @api.multi
+    def _write(self, vals):
+        # Manual computing of 'completed' field (because depending of 'completion' computed stored field)
+        if 'completion' in vals and vals['completion'] >= 100:
+            vals.update({'completed': True})
+            partner_has_completed = {channel_partner.partner_id.id: channel_partner.channel_id for channel_partner in self}
```

Comment:
I would call super then search all suers and update their karma. I also think you will ran into issues without any sudo if current user is not admin.